zerocopy-derive/src/lib.rs

```diff
@@ -552,39 +552,38 @@ fn impl_block<D: DataExt>(
         quote!()
     };
 
-    let size_check_body = match (field_types.is_empty(), padding_check) {
-        (true, _) | (false, PaddingCheck::None) => quote!(),
-        (false, PaddingCheck::Struct) => quote!(
-            const _: () = {
-                trait HasPadding<const HAS_PADDING: bool> {}
-                fn assert_no_padding<T: HasPadding<false>>() {}
-
-                const COMPOSITE_TYPE_SIZE: usize = ::core::mem::size_of::<#type_ident>();
-                const SUM_FIELD_SIZES: usize = 0 #(+ ::core::mem::size_of::<#field_types>())*;
-                const HAS_PADDING: bool = COMPOSITE_TYPE_SIZE > SUM_FIELD_SIZES;
-                impl HasPadding<HAS_PADDING> for #type_ident {}
-                let _ = assert_no_padding::<#type_ident>;
-            };
-        ),
-        (false, PaddingCheck::Union) => quote!(
-            const _: () = {
-                trait FieldsAreSameSize<const FIELDS_ARE_SAME_SIZE: bool> {}
-                fn assert_fields_are_same_size<T: FieldsAreSameSize<true>>() {}
-
-                const COMPOSITE_TYPE_SIZE: usize = ::core::mem::size_of::<#type_ident>();
-                const FIELDS_ARE_SAME_SIZE: bool = true
-                    #(&& (::core::mem::size_of::<#field_types>() == COMPOSITE_TYPE_SIZE))*;
-                impl FieldsAreSameSize<FIELDS_ARE_SAME_SIZE> for #type_ident {}
-                let _ = assert_fields_are_same_size::<#type_ident>;
-            };
-        ),
-    };
+    match (field_types.is_empty(), padding_check) {
+        (true, _) | (false, PaddingCheck::None) => (),
+        (false, PaddingCheck::Struct) => {
+            let fields = field_types.iter();
+            // `parse_quote!` doesn't parse macro invocations in const generics
```

Comment:
Can you add to this comment showing an example of what the below code generates? It's hard to see what it's doing on its own. Same in the last match arm.